adam-core/src/main/scala/org/bdgenomics/adam/converters/FastqRecordConverter.scala

```diff
@@ -41,6 +41,114 @@ import scala.collection.JavaConversions._
 private[adam] class FastqRecordConverter extends Serializable with Logging {
 
   /**
+   * Parse 4 lines at a time
+   * @see parseReadPairInFastq
+   * *
+   */
+  private[converters] def readNameSuffixAndIndexOfPairMustMatch(readName: String,
+                                                                isFirstOfPair: Boolean): Unit = {
+    val firstReadSuffix = """[/ +_]1$""".r
+    val secondReadSuffix = """[/ +_]2$""".r
+
+    val isSecondOfPair = !isFirstOfPair
+
+    val match1 = firstReadSuffix.findAllIn(readName)
+    val match2 = secondReadSuffix.findAllIn(readName)
+
+    if (match1.nonEmpty && isSecondOfPair)
+      throw new IllegalArgumentException(
+        s"Found read name $readName ending in ${match1.next} despite first-of-pair flag being set")
+    else if (match2.nonEmpty && isFirstOfPair)
+      throw new IllegalArgumentException(
+        s"Found read name $readName ending in ${match2.next} despite second-of-pair flag being set")
+    // else, readName doesn't really tell whether it's first or second of pair, assumed to match
+  }
+
+  private[converters] def parseReadInFastq(input: String,
+                                           setFirstOfPair: Boolean = false,
+                                           setSecondOfPair: Boolean = false,
+                                           stringency: ValidationStringency = ValidationStringency.STRICT): (String, String, String) = {
+    // since it's a private method, simple require call is ok without detailed error message
+    require(!(setFirstOfPair && setSecondOfPair))
+
+    val lines = input.split('\n')
+    require(lines.length == 4,
+      s"Input must have 4 lines (${lines.length.toString} found):\n${input}")
+
+    val readName = lines(0).drop(1)
+    if (setFirstOfPair || setSecondOfPair) readNameSuffixAndIndexOfPairMustMatch(readName, setFirstOfPair)
+
+    val suffix = """([/ +_]1$)|([/ +_]2$)""".r
```

Comment:
I wonder if this regex and those above on line 50 and 51 might be combined as static private fields so that the two methods don't get out of sync